### PR TITLE
refactor: use package server/api to register supernode's API

### PR DESCRIPTION
**Pull request by lowzj**
Signed-off-by: lowzj <zj3142063@gmail.com>

### Ⅰ. Describe what this PR did

use package server/api to register supernode's APIs

### CODE REVIEW

File: supernode/server/api/utils.go [hunk 1]
```diff
@@ -52,26 +52,28 @@ func ParseJSONRequest(req io.Reader, target interface{}, validator ValidateFunc)
...
 }
 
-// EncodeResponse encodes response in json.
-// The response body is empty if the data is nil or empty value.
-func EncodeResponse(w http.ResponseWriter, code int, data interface{}) error {
+// SendResponse encodes response in json.
+//
+// TODO:
+//  Should the response body should be empty if the data is nil or empty
+//  string? Now it's incompatible with the client.
```
Comment (zhouhaibing089) [inline supernode/server/api/utils.go:59]: I don't think we should. If the data is nil/empty, it basically means there is an error somehow, in that case, the `code` should reflect which error it is, and data should be used to describe what error it is.

File: supernode/server/api/utils.go [hunk 2]
```diff
@@ -52,26 +52,28 @@ func ParseJSONRequest(req io.Reader, target interface{}, validator ValidateFunc)
...
+	return enc.Encode(data)
 }
 
 // HandleErrorResponse handles err from server side and constructs response
 // for client side.
 func HandleErrorResponse(w http.ResponseWriter, err error) {
 	switch e := err.(type) {
 	case *errortypes.HTTPError:
-		_ = EncodeResponse(w, e.Code, errResp(e.Code, e.Msg))
+		_ = SendResponse(w, e.Code, errResp(e.Code, e.Msg))
```
Comment (zhouhaibing089) [inline supernode/server/api/utils.go:73]: This is odd, The error should not be ignored silently, right?

File: supernode/server/api/utils.go (hunk 1 above)
Comment (lowzj) [reply supernode/server/api/utils.go:60]: Ok, maybe we should provide a method to send header only.

File: supernode/server/api/utils.go (hunk 2 above)
Comment (lowzj) [reply supernode/server/api/utils.go]: done

File: supernode/server/router.go [hunk 3]
```diff
@@ -50,91 +98,46 @@ func initRoute(s *Server) *mux.Router {
...
-			HandleErrorResponse(w, err)
-		}
-	}
+func handleMetrics(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	promhttp.Handler().ServeHTTP(rw, req)
+	return nil
 }
 
-// EncodeResponse encodes response in json.
+// SendResponse encodes response in json.
```
Comment (Starnop) [inline supernode/server/router.go:140]: s/ SendResponse/ EncodeResponse

File: supernode/server/api/api.go [hunk 4]
```diff
@@ -47,6 +47,9 @@ func newCategory(name, prefix string) *category {
 	apiCategories[name] = &category{
 		name:   name,
 		prefix: prefix,
+		handlerSpecs: []*HandlerSpec{
+			listHandler(name),
```
Comment (Starnop) [inline supernode/server/api/api.go:51]: why should we add a default list handler here?

Comment (lowzj) [reply supernode/server/api/api.go:51]: I can help up to get all the available APIs provided by supernode.

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**